cpp/src/parquet/encoding.cc

```diff
@@ -4055,4 +4055,57 @@ std::unique_ptr<Decoder> MakeDictDecoder(Type::type type_num,
 }
 
 }  // namespace detail
+
+bool IsParquetVersionAtLeast2_0(ParquetVersion::type parquet_version) {
+  switch (parquet_version) {
+    case ParquetVersion::PARQUET_1_0:
+      return false;
+    case ParquetVersion::PARQUET_2_4:
+    case ParquetVersion::PARQUET_2_6:
+      return true;
+    default:
+      return parquet_version > ParquetVersion::PARQUET_1_0;
+  }
+  return false;
+}
+
+// Informed heavily by https://github.com/apache/parquet-format/blob/master/Encodings.md
+// and
+// https://github.com/apache/arrow-rs/blob/a15adf66a8212fdb3a0222a32130ce70cfc28cf5/parquet/src/column/writer/mod.rs#L1183
```

Comment:
BTW, it seems intuitive to fall back to BYTE_STREAM_SPLIT for FLOAT/DOUBLE. arrow-rs does not implement BYTE_STREAM_SPLIT so it does not appear there (CMIW).